src/middlewared/middlewared/plugins/virt/instance.py

```diff
@@ -0,0 +1,445 @@
+
+        def unsupported():
+            self.logger.warning('Proxy device not supported by API, skipping.')
+            return None
+
+        match incus['type']:
+            case 'disk':
+                device['dev_type'] = 'DISK'
+                device['source'] = incus.get('source')
+                device['destination'] = incus.get('path')
+            case 'nic':
+                device['dev_type'] = 'NIC'
+                device['network'] = incus.get('network')
+            case 'proxy':
+                device['dev_type'] = 'PROXY'
+                # For now follow docker lead for simplification
+                # only allowing to bind on host (host -> container)
+                if incus.get('bind') == 'instance':
+                    return unsupported()
+
+                proto, addr, ports = incus['listen'].split(':')
+                if proto == 'unix' or '-' in ports or ',' in ports:
+                    return unsupported()
+
+                device['source_proto'] = proto.upper()
+                device['source_port'] = int(ports)
+
+                proto, addr, ports = incus['connect'].split(':')
+                if proto == 'unix' or '-' in ports or ',' in ports:
+                    return unsupported()
+
+                device['dest_proto'] = proto.upper()
+                device['dest_port'] = int(ports)
+            case 'tpm':
+                device['dev_type'] = 'TPM'
+                device['path'] = incus.get('path')
+                device['pathrm'] = incus.get('pathrm')
+            case 'usb':
+                device['dev_type'] = 'USB'
+                if incus.get('busnum') is not None:
+                    device['bus'] = int(incus['busnum'])
+                if incus.get('devnum') is not None:
+                    device['dev'] = int(incus['devnum'])
+                if incus.get('productid') is not None:
+                    device['product_id'] = incus['productid']
+                if incus.get('vendorid') is not None:
+                    device['vendir_id'] = incus['vendorid']
+            case 'gpu':
+                device['dev_type'] = 'USB'
+                device['id'] = incus['id']
+                device['gpu_type'] = incus['gputype'].upper()
+                match incus['gputype']:
+                    case 'physical':
+                        pass
+                    case 'mdev':
+                        pass
+                    case 'mig':
+                        device['mig_uuid'] = incus['mig.uuid']
+                    case 'sriov':
+                        pass
+            case _:
+                return unsupported()
+
+        return device
+
+    async def __device_to_incus(self, instance_type: str, device: dict[str, Any]) -> dict[str, Any]:
+        new = {}
+
+        match device['dev_type']:
+            case 'DISK':
+                new['type'] = 'disk'
+                source = device.get('source') or ''
+                if not source.startswith(('/dev/zvol/', '/mnt/')):
```

Comment:
Do we need to exclude `/mnt/.ix-apps`?

Reply:
Do you have any validation that the zvol isn't in use by another part of the product?

Reply:
Paths can be shared to containers, e.g. sharing media to a plex container

I will add more validation.